### PR TITLE
playbook for installing and running cardiff

### PR DESCRIPTION
**Pull request by MoteHue**

### CODE REVIEW

File: etc/kayobe/ansible/cardiff-run.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+---
+- name: Cardiff run
+  hosts: localhost
+  gather_facts: no
+  tags:
+    - cardiff
+  vars:
+    venv: "~/venvs/cardiff-venv"
+    input_dir: "{{ lookup('env', 'PWD') }}/overcloud-introspection-data"
+    output_dir: "{{ lookup('env', 'PWD') }}/review"
```
Comment (markgoddard) [inline etc/kayobe/ansible/cardiff-run.yml]: More specific name? `cardiff-review`?

File: etc/kayobe/ansible/cardiff-run.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    cardiff_pattern: ".*.eval" # Uses regex
+  tasks:
+    - name: Install dependencies
+      pip:
+        virtualenv: "{{ venv }}"
+        name:
+          - git+https://github.com/stackhpc/cardiff
+          #- /home/matt/Documents/cardiff
```
Comment (markgoddard) [inline etc/kayobe/ansible/cardiff-run.yml]: Remove me :)

File: etc/kayobe/ansible/cardiff-run.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+          - pyvis.network
+        state: latest
+
+    - name: Create data directory
+      file:
+        path: '{{ output_dir }}/data'
+        state: directory
+
+    - name: Extract data
+      shell:
```
Comment (markgoddard) [inline etc/kayobe/ansible/cardiff-run.yml]: nit: could use `command`

File: etc/kayobe/ansible/cardiff-run.yml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        cmd: >
+          {{ venv }}/bin/m2-extract {{ input_dir }}/*.json --output_dir {{ output_dir }}/data
+
+    - name: Create review directory
+      file:
+        path: '{{ output_dir }}/results'
+        state: directory
+
+    - name: Process data
+      shell:
```
Comment (markgoddard) [inline etc/kayobe/ansible/cardiff-run.yml]: nit: could use command

File: etc/kayobe/ansible/cardiff-run.yml [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        cmd: >
+          {{ venv }}/bin/cardiff
+          -I ipmi
+          -p '{{ output_dir }}/data/extra-hardware/{{ cardiff_pattern }}'
+          -o '{{ output_dir }}/results'
+          --visualise
+        #  --visualise_dataless
```
Comment (markgoddard) [inline etc/kayobe/ansible/cardiff-run.yml]: Are there various options that may be useful here? Could make it a variable:

```
cardiff_opts: "--visualise"
```

File: etc/kayobe/ansible/cardiff-run.yml (hunk 3 above)
Comment (MoteHue) [reply etc/kayobe/ansible/cardiff-run.yml]: These two commands need to use shell for the filename expansions

File: etc/kayobe/ansible/cardiff-run.yml (hunk 5 above)
Comment (MoteHue) [reply etc/kayobe/ansible/cardiff-run.yml]: Decided the dataless option was unnecessary, have removed it.

File: etc/kayobe/ansible/cardiff-run.yml (hunk 3 above)
Comment (markgoddard) [reply etc/kayobe/ansible/cardiff-run.yml]: ack

**Review by markgoddard — DISMISSED** [review]

**Review by markgoddard — APPROVED** [review]

**Pull request merged**